Add CodeOps script for standardising GitHub settings across Orgs/Repos

JamesDawson opened the pull request:
This gives us a mechanism for applying certain repository-level settings across our GitHub organisations.

This version implements two such policies:
1. Applies the `delete_branch_on_merge` setting (ref: https://docs.github.com/en/rest/reference/repos#create-an-organization-repository)
2. Ensures branch protection for the default branch, enforcing the `strict` and `enforce_admins` settings (ref: https://docs.github.com/en/rest/reference/repos#update-branch-protection)

Expanding policies should be a case of creating additional scripts as per the notes here: `codeops-scripts/github-settings-definitions/README.md`

JamesDawson committed:
WIP

JamesDawson committed:
Finalised scripts

Moved 'setting' definitions into separate scripts for ease of maintenance
Made branch protection policy lookup the repo's default branch
Added README for extending settings policies

JamesDawson committed:
fix broken test due to not mocking-out calls to nuget.org

JamesDawson committed:
Updating workflows with GitVersion fixes

JamesDawson committed:
Removed 'delete_branch_on_merge' prautoflow script

JamesDawson committed:
Trimmed-down github rest api test data

HowardvanRooijen commented on codeops-scripts/apply-github-settings.ps1:
```diff
@@ -0,0 +1,198 @@
+#
+# Helper functions
+#
+function _getAllOrgReposWithDefaults {
```
Does this only deal with public repos?

JamesDawson replied:
It will find all repos public or private, although the branch protection policy can only be applied to public repos.

What are your thoughts on whether to include the 'endjin' org?

HowardvanRooijen commented on codeops-scripts/apply-github-settings.ps1:
```diff
@@ -0,0 +1,198 @@
+function _getAllOrgs {
+    # returns all the orgs that we want included in this process
+    @(
+        "ais-dotnet"
```
If this was driven by a yaml file - this could become more reusable for external folk

JamesDawson replied:
Agreed. Originally I'd thought to derive the organisations from the existing YAML files, but parked it to make some progress.

Having got the other merging functionality working, I'll take another look.

JamesDawson committed:
List of GitHub orgs now aggregated from the YAML config.

Added another test to cover edge case discovered whilst making the above changes.

JamesDawson committed:
Update live repo config.

Removed githubSettings that are now applied via the standard policy.

JamesDawson committed:
Added initial workflow for running 'apply_github_setttings' codeops script

JamesDawson committed:
Merge branch 'master' into feature/github-settings

JamesDawson committed:
Bump Endjin.CodeOps module

MikeEvansLarah commented on codeops-scripts/github-settings-definitions/default_branch_protection.ps1:
```diff
@@ -0,0 +1,66 @@
+            }
+        }
+
+        # We can only set branch protection on public repos
```
How come? Is this a free vs paid restriction?

JamesDawson replied:
Yes, that's right

JamesDawson committed:
temporary exemption for new 'test-data-gen' repo
